### PR TITLE
fix possible end() iterator incrementation bugs

### PR DESCRIPTION
**Pull request by vogel**
Fixes heap-buffer-overflow (READ of size 1) in cpptoml::parser::consume_whitespace()

Signed-off-by: Rafał Malinowski <rafal.przemyslaw.malinowski@gmail.com>

### CODE REVIEW

File: lib/util/cpptoml.h [hunk 1]
```diff
@@ -1108,6 +1111,11 @@ namespace cpptoml
             // ugh---have to read the first value to determine array type...
             skip_whitespace_and_comments(it, end);
 
+            if (it == end)
+            {
+                throw_parse_exception("Unable to parse array");
```
Comment (zbyszkol) [inline lib/util/cpptoml.h]: Put more descriptive error message, e.g. unexpected end of an array...

File: lib/util/cpptoml.h [hunk 2]
```diff
@@ -1148,8 +1156,15 @@ namespace cpptoml
                 else
                     throw_parse_exception("Arrays must be homogeneous");
                 skip_whitespace_and_comments(it, end);
+
+                if (it == end)
+                {
+                    throw_parse_exception("Unable to parse array");
```
Comment (zbyszkol) [inline lib/util/cpptoml.h]: Put more descriptive error message, e.g. unexpected end of an array...

File: lib/util/cpptoml.h [hunk 3]
```diff
@@ -1166,6 +1181,12 @@ namespace cpptoml
             {
                 arr->array().push_back(parse_array(it, end));
                 skip_whitespace_and_comments(it, end);
+
+                if (it == end)
+                {
+                    throw_parse_exception("Unable to parse array");
```
Comment (zbyszkol) [inline lib/util/cpptoml.h]: Put more descriptive error message, e.g. unexpected end of an array...

**Comment by vogel** [comment]
I've found out that these exceptions were not needed - conditions was always false, as new assert shows now.

**Comment by MonsieurNicolas** [comment]
We should not make arbitrary changes to vendored code; this was already fixed in upstream.
See https://github.com/skystrife/cpptoml

**Pull request closed without merging**